python/metatensor-operations/metatensor/operations/_dispatch.py

```diff
@@ -279,6 +279,20 @@ def eye_like(array, size: int):
         raise TypeError(UNKNOWN_ARRAY_TYPE)
 
 
+def get_device(array) -> Optional[torch_device]:
+    """
+    Returns the device of the array if it is a
+    ``torch.Tensor``, or None if it is a ``numpy.ndarray``.
+    """
+
+    if isinstance(array, TorchTensor):
+        return array.device
+    elif isinstance(array, np.ndarray):
+        return None
```

Comment:
most of the code until now returned `"cpu"` for the device of numpy arrays, should this do the same?